TryAtSoftware.Extensions.Collections/FluidDictionary.cs

```diff
@@ -0,0 +1,113 @@
+namespace TryAtSoftware.Extensions.Collections;
+
+using System;
+using System.Collections.Generic;
+using TryAtSoftware.Extensions.Collections.Interfaces;
+
+public class FluidDictionary<TKey> : IFluidDictionary<TKey>
+{
+    private Dictionary<TKey, object> dictionary;
+
+    public FluidDictionary()
+    {
+        dictionary = new Dictionary<TKey, object>();
+    }
+
+    /// <summary>
+    /// Sets the value of type T associated with the specified key in the dictionary.
+    /// </summary>
+    /// <typeparam name="T">The type of the value to set.</typeparam>
+    /// <param name="key">The key to set the value for.</param>
+    /// <param name="value">The value to set.</param>
+    /// <returns>True if the value is successfully set, false otherwise.</returns>
+    public bool Set<T>(TKey key, T value)
+    {
+        if (key == null)
+            throw new ArgumentNullException(nameof(key));
+
+        dictionary[key] = value;
+        return true;
+    }
+
+    /// <summary>
+    /// Attempts to get the value associated with the specified key in the dictionary.
+    /// </summary>
+    /// <typeparam name="T">The type of the value to retrieve.</typeparam>
+    /// <param name="key">The key whose value to get.</param>
+    /// <param name="value">When this method returns, contains the value associated with the specified key,
+    /// if the key is found; otherwise, the default value for the type T.</param>
+    /// <returns>True if the key is found and the value is of type T, false otherwise.</returns>
+    public bool TryGetValue<T>(TKey key, out T value)
+    {
+        if (dictionary.TryGetValue(key, out var objValue) && objValue is T)
```

Comment:
I think you should be able to utilize pattern matching at max and avoid the unnecessary cast on line 44.